src/automations/components/AutomationActionDescriptionSendNotification.vue

```diff
@@ -0,0 +1,52 @@
+<template>
+  <div class="automation-action-description-send-notification" v-bind="$attrs">
+    Send {{ indefiniteArticle() }} {{ blockDocument?.blockType.name.toLocaleLowerCase() ?? '' }} notification using
+    <BlockIconText :block-document-id="blockDocumentId" />
+    {{ recipients }}
+  </div>
+</template>
+
+<script lang="ts" setup>
+  import { computed } from 'vue'
+  import { AutomationActionSendNotification } from '@/automations/types/actions'
+  import { BlockIconText } from '@/components'
+  import { useBlockDocument } from '@/compositions'
+
+  defineOptions({
+    inheritAttrs: false,
+  })
+
+  const props = defineProps<{
+    action: AutomationActionSendNotification,
+  }>()
+
+  const { blockDocument } = useBlockDocument(props.action.blockDocumentId)
+
+  const {
+    blockDocumentId,
+  } = props.action
```

Comment:
Since props is reactive if we destructure like this the reactivity of blockDocumentId will be broken. Meaning if the action being passed to this component changes, the blockDocumentId won't update to reflect that the action changed. 

Either 
a. Use a computed 
b. Use action.blockDocumentId directly (preferred when just using this in the template)

Reply:
Ah that makes sense! Definitely something I'm less familiar with in Vue but the concept rings true.